fabric-events-interaction-v0/src/main/java/net/fabricmc/fabric/mixin/event/interaction/BlockMixin.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.event.interaction;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+import net.fabricmc.fabric.api.event.BlockBreakEffectsCallback;
+
+@Mixin(Block.class)
+public class BlockMixin {
+	@Redirect(at = @At(value = "INVOKE", target = "Lnet/minecraft/world/World;syncWorldEvent(Lnet/minecraft/entity/player/PlayerEntity;ILnet/minecraft/util/math/BlockPos;I)V"), method = "onBreak")
+	public void onBreak(World world, PlayerEntity player, int eventId, BlockPos pos, int data, World world2, BlockPos pos2, BlockState state, PlayerEntity player2) {
```

Comment:
This injection does not cover every single usecase. Since blocks may override `onBreak` at will some implementations of this may never fire and particles will be generated.

You can't redirect the callsite in the client or server player interaction managers either otherwise some logic like piglin brain notification fail.

I think we need to think about how to handle this without hardcoding ourselves to hell here.

Reply:
If a block override ``onBreak`` the  the break effects won't show anyways. IMO it is best to just not fire the event in that situation, since the modder is just replacing the effects somewhere else. All vanilla blocks call ``super`` except fire, because fire doesn't show break block effects. If it doesn't show break block effects, why would you fire the event?